python/aitubertalk/rooms/room_module.py

```
@@ -0,0 +1,357 @@
+            self._client.logger.info(f"Retrieved room: {room.name}")
+            return room
+            
+        except AITuberTalkError:
+            raise
+        except Exception as e:
+            self._client.logger.error(f"Room retrieval failed: {e}")
+            raise AITuberTalkError(
+                code=ErrorCode.NETWORK_ERROR,
+                message=f"Failed to get room: {e}",
+                retryable=True
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)